### DISCUSSION TITLE
Why does "back_populates" move an transient object to pending?

### DISCUSSION THREAD

**Discussion by SergeNasr** [sqlalchemy/sqlalchemy#6892] [Usage Questions]
Hello!

I am trying to understand the following behavior.

Assume I have the following two models:
```
class Parent(db.Model):
    ...

class Child(db.Model):
    ...
    parent = db.relationship(Parent, back_populates="children")
```

```
# I first create a transient Parent object
parent = Parent()

insp = inspect(parent)
insp.trasient # This prints True

# I then create a transient Child object
child = Child()

insp = inspect(child)
insp.trasient # This prints True

# But when I add "parent" to the relationship of "child", both "parent" and "child" move to pending state
child.parent = parent
insp = inspect(child)
insp.trasient  # This prints False
insp.pending # This prints True!!

insp = inspect(parent)
insp.trasient  # This prints False
insp.pending # This prints True!!
```

Has anybody seen this behavior? I can't seem to figure out from SQLA's code why this happens

**Comment by zzzeek** [comment]
this is not possible if you don't have a `Session()` involved.   if you are using some extension that implicitly associates objects with sessions, i think perhaps flask-sqlalchemy might do something like this, that would be the reason.

**Reply by SergeNasr** [reply]
Yes there is a session and I am using `flask-sqlalchemy` 
Where can I read more about this behavior?

**Reply by zzzeek** [reply]
i dont know, looking at their current docs they seem to document explicit use of session.add().      it was a much older pattern that I'm not sure they have implemented anymore.

in your inspect you can call upon `.session` to get the current Session that they are associated with, see if that's the flask-sqlalchemy session.

you can also attach an event hook for "before_attach" https://docs.sqlalchemy.org/en/14/orm/events.html?highlight=sessionevents%20attach#sqlalchemy.orm.SessionEvents.before_attach and that will indicate exactly when the attachment is happening.
